### ISSUE TITLE
HLint should catch adjacent misspellings

### ISSUE THREAD

**Issue by moldux-dartus** [ndmitchell/hlint#1429]
HLint should alert the user if two function definitions are very similar and only separated by a newline. Example

```
changeCode :: String -> Lock -> Lock\n
changeCode newCode (Open code) = Open newCode
changecode _ l = l
```

This compiles without error but produces the following error when tested:

```*** Failed! (after 1 test):
Exception:
  Set9a.hs:150:1-45: Non-exhaustive patterns in function changeCode
isOpen (open "1234" (changeCode "0000" aLock))
```
This error is an undesirable runtime exception that would ideally be caught by a linter. Although there are legitimate reasons to have very similiar function definitions directly adjacent, it is uncommon enough that it should be flagged as potentially undesirable behavior by HLint.

**Comment by zliu41** [comment]
GHC by default compiles it without warnings, but
- you'll get the incomplete pattern warning with `-W`, which almost all non-test code should use imo
- if the module also has an explicit export list, which is also good practice, then you'll also get the "defined but not used" warning
- with `-Wall` you'll also get the "missing signature" warning.

Given these I do not think HLint should try to catch this, which is quite tricky.

**Comment by moldux-dartus** [comment]
> GHC by default compiles it without warnings, but
> 
> * you'll get the incomplete pattern warning with `-W`, which almost all non-test code should use imo
> * if the module also has an explicit export list, which is also good practice, then you'll also get the "defined but not used" warning
> * with `-Wall` you'll also get the "missing signature" warning.
> 
> Given these I do not think HLint should try to catch this, which is quite tricky.

You're right, since I was working with a test suite an
